cartoframes/data/observatory/dataset.py

```diff
@@ -1,34 +1,76 @@
 from __future__ import absolute_import
 
-from .entity import SingleEntity, EntitiesList
+from .entity import CatalogEntity
 from .repository.dataset_repo import get_dataset_repo
 from .repository.variable_repo import get_variable_repo
 from .repository.variable_group_repo import get_variable_group_repo
 
 _DATASET_ID_FIELD = 'id'
 
 
-class Dataset(SingleEntity):
+class Dataset(CatalogEntity):
 
     id_field = _DATASET_ID_FIELD
     entity_repo = get_dataset_repo()
 
-    @classmethod
-    def _get_entities_list_class(cls):
-        return Datasets
-
     def variables(self):
-        return get_variable_repo().get_by_dataset(self._get_id())
+        return get_variable_repo().get_by_dataset(self.id)
 
     def variables_groups(self):
-        return get_variable_group_repo().get_by_dataset(self._get_id())
+        return get_variable_group_repo().get_by_dataset(self.id)
 
+    @property
```

Comment:
I think we can define __getattr__ at CatalogEntity and forget about setting all the properties by hand.

Not a priority anyway.

Reply:
I intentionally let those property methods unchanged at this point. We still have to decide the long-term behavior for the "id-like" fields and also we're currently relying on those property methods to rename some of the columns. So I would prefer to address that discussion in a separated PR for when we have a stable version of the hierarchy :)